File: sdk/java-sdk/src/main/java/com/evernym/verity/sdk/protocols/presentproof/common/ProposedAttribute.java
package com.evernym.verity.sdk.protocols.presentproof.common;

import com.evernym.verity.sdk.utils.AsJsonObject;
import org.json.JSONObject;

public class ProposedAttribute implements AsJsonObject {
    final JSONObject data;


    /**
     * Constructs the ProposedAttribute object
     * @param name the attribute name
     * @param credDefId credential from which attribute is obtained.
     * @param value value of the attribute.
     */
    public ProposedAttribute(String name, String credDefId, String value){
        this.data = new JSONObject()
                .put("name", name)
                .put("cred_def_id", credDefId)
                .put("value", value);
    }

    /**
     * Convert this object to a JSON object
     *
     * @return this object as a JSON object
     */
    @Override
    public JSONObject toJson() {
        return data;
    }
}

File: sdk/dotnet-sdk/src/VeritySDK/Protocols/PresentProof/Common/ProposedAttribute.cs
using System.Json;
using VeritySDK.Utils;

namespace VeritySDK.Protocols.PresentProof
{
    /// <summary>
    /// A holder for the ProposedAttribute
    /// </summary>
    public class ProposedAttribute : AsJsonObject
    {
        private JsonObject _data = new JsonObject();

        public JsonObject data { get { return _data; } }

        /// <summary>
        /// Constructs the ProposedAttribute object
        /// </summary>
        /// <param name="name">the attribute name</param>
        /// <param name="credDefId">credential from which attribute is obtained.</param>
        /// <param name="value">value of the attribute</param>
        public ProposedAttribute(string name, string credDefId, string value)
        {
            data.Add("name", name);
            data.Add("cred_def_id", credDefId);
            data.Add("value", value);
        }

        /// <summary>
        /// Convert this object to a JSON object 
        /// </summary>
        /// <returns>this object as a JSON object</returns>
        public JsonObject toJson()
        {
            return data;
        }
    }
}